pkg/proxy/proxy_test.go

```
@@ -0,0 +1,96 @@
+	err = proxy.StreamRequest()
+	if err != nil {
+		t.Errorf("StreamRequest failed: %v", err)
+	}
+
+	resp := <-respCh
+	if resp == nil {
+		t.Errorf("Failed to read response")
+	}
+
+	// Check the response status code
+	if resp.StatusCode != http.StatusOK {
```

golangci-lint (1.62.2)

SA5011: possible nil pointer dereference

(staticcheck)